objects/src/state.rs

```diff
@@ -0,0 +1,84 @@
+use crate::{AccountId, BlockHeader, ChainMmr, Digest, Vec};
+use crypto::merkle::{DefaultMerkleStore, TieredSmt};
+
+/// Structure chain data, used to build necessary openings and to construct [BlockHeader]s.
+#[derive(Default, Debug, Clone)]
+pub struct State {
+    /// An append-only structure used to represent the history of blocks produced for this chain.
+    chain: ChainMmr,
+
+    /// Storage for the historical state of the chain.
+    ///
+    /// This store contains historical data for accounts, nullifiers, notes, and batches.
+    data: DefaultMerkleStore,
+
+    /// History of produced blocks.
+    blocks: Vec<BlockHeader>,
+
+    /// Tree containing the latest [Nullifier] tree.
+    nullifier_db: TieredSmt,
+
+    /// The latest block being produced
+    pending: PendingBlock,
+}
```

Comment:
A few comments on this:

1. We should probably have a vector of all current account states here - e.g., `Vec<Account>`. Eventually, we'll replace it with something that emulates account DB - but for now just a vector should do. For the user standpoint, it might be useful to be able to get reference to an account by either ID or maybe just index.
2. Similarly, we should probably have a vector of all notes in the chain - e.g., `Vec<Note>` and then allow the user to get a note either by hash or by index (or maybe something else).
3. Having `PendingBlock` be a part of this struct feels a bit weird to me. When this struct is instantiated from a file, I'm not sure we need to have a "pending block". So, I wonder if this part can be refactored so that we do block construction externally.
4. I think having `data` as generic storage is fine for now but I also think that it's very likely we'll have to refactor this in the future to make separation between blocks much more clear. One reason for this is that we might want to get a list of all notes created in a given block, or a list of all accounts updated in a given block etc. And I think currently, this might be non-trivial.

Lastly, a minor nit: I wonder if the name `State` is right for this struct. It feel a bit too generic. Maybe something like `Chain` or `Mock` or `MockChain` would be better?